src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,245 @@
+        release: str,
+        hf_token: str,
+    ) -> None:
+        super().__init__(
+            ctx=ctx,
+            model=model,
+            model_dir=model_dir,
+            upload_dest=upload_dest,
+            release=release,
+        )
+        self.hf_token = hf_token
+        self.local_model_path = ""
+
+    def upload(self):
+        """
+        Upload specified model to Hugging Face
+        """
+
+        # HF token check
+        if self.hf_token == "":
+            raise ValueError(
+                """HF_TOKEN var needs to be set in your environment to upload HF Model.
+                Alternatively, the token can be passed with --hf-token flag.
+                The HF Token is used to authenticate your identity to the Hugging Face Hub."""
+            )
+
+        click.echo(
+            f"Uploading model to Hugging Face:\n{DEFAULT_INDENT}Model: {self.model}\n{DEFAULT_INDENT}Destination: {self.upload_dest}@{self.release}\n"
+        )
+
+        # set logger
+        if self.ctx.obj is not None:
+            hf_logging.set_verbosity(self.ctx.obj.config.general.log_level.upper())
+
+        # determine if model is path or name
+        if os.path.exists(self.model):
+            self.local_model_path = Path(self.model)
+            self.model = self.model.split("/")[-1]
+        else:
+            self.local_model_path = os.path.join(self.model_dir, self.model)
+            # throw an error if the built path is invalid
+            if not os.path.exists(self.local_model_path):
+                click.secho(
+                    f"Couldn't find model at {self.local_model_path} - are you sure it exists?",
+                    fg="red",
+                )
+                raise click.exceptions.Exit(1)
+
+        # upload either safetensors or gguf, if valid
+        if is_model_safetensors(self.local_model_path):
+            self.upload_safetensors()
+        elif is_model_gguf(self.local_model_path):
+            self.upload_gguf()
+        else:
+            click.secho(
+                f"Local model path {self.local_model_path} is a valid path, but is neither safetensors nor a GGUF - cannot upload to Hugging Face",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def upload_gguf(self) -> None:
+        try:
+            hf_api = HfApi()
+            hf_api.upload_file(
+                path_or_fileobj=self.local_model_path,
+                path_in_repo=self.model,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading GGUF model succeeded!",
+            fg="green",
+        )
```

Comment:
Can we print out the `upload_file` response? It appears that the `upload_file` function returns the URL to the upload: https://github.com/huggingface/huggingface_hub/blob/v0.26.5/src/huggingface_hub/hf_api.py#L4677L4681

We can confirm that the URL gets printed out by looking at Hugging Face's unit tests: https://github.com/huggingface/huggingface_hub/blob/main/tests/test_hf_api.py

Adding this kind of logging can help users triage issues related to uploading (like if somehow there is an uncaught exception on the Hugging Face end), and let them confirm where it was uploaded to.

```suggestion
    def upload_gguf(self) -> None:
        try:
            hf_api = HfApi()
            resp = hf_api.upload_file(
                path_or_fileobj=self.local_model_path,
                path_in_repo=self.model,
                repo_id=self.upload_dest,
                revision=self.release,
                token=self.hf_token,
            )
        except Exception as exc:
            click.secho(
                f"\nUploading GGUF model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
                fg="red",
            )
            raise click.exceptions.Exit(1)
        click.secho(
            f"\nUploading GGUF model succeeded! Hugging Face response:\n{resp}",
            fg="green",
        )
```